compaction/compaction_manager.cc

```diff
@@ -1134,16 +1134,23 @@ future<> compaction_manager::stop() {
     if (auto cm = std::exchange(_task_manager_module, nullptr)) {
         co_await cm->stop();
     }
-    if (_state != state::none) {
-        co_return co_await std::move(*_stop_future);
+    if (_stop_future) {
+        co_await std::move(*_stop_future);
     }
 }
 
-future<> compaction_manager::really_do_stop() {
+future<> compaction_manager::really_do_stop() noexcept {
     cmlog.info("Asked to stop");
     // Reset the metrics registry
     _metrics.clear();
-    co_await stop_ongoing_compactions("shutdown");
+    try {
+        co_await stop_ongoing_compactions("shutdown");
+    } catch (...) {
+        cmlog.warn("Ignored error while stopping compaction tasks: {}", std::current_exception());
+    }
+    if (!_tasks.empty()) {
+        on_fatal_internal_error(cmlog, format("{} tasks still exist after being stopped", _tasks.size()));
```

Comment:
This contradicts with the `catch (...)` above: we want to ignore errors during stop compaction, at the same time you assert that stop compaction finished. If there was an error, stop compaction will not have succeeded and it is not guaranteed that `_tasks` is empty.

Reply:
Ok, I see that `stop_tasks()` should call `stop()` on all tasks, even if some of them fail. So I guess this is fine. We just need to fix the race.

Reply:
maybe stop compactions should assert _tasks.empty() when done

Reply:
I think it should guarantee that.
The fact that a task completed with an error doesn't mean it wasn't properly stopped.
We should document that stop_ongoing_compactions and stop_tasks always succeed, and any error the may return, just represents the task completion status, not the status of stopping them.